app/controllers/rentals_controller.rb

```diff
@@ -36,14 +36,14 @@ def show; end
 
   # GET /rentals/cost?end_time=time&start_time=time&item_types=...
   def cost
-    _params = cost_params
+    cost_params = params.permit(:start_time, :end_time, item_types: [])
```

Comment:
If you change this to `params.require` like Dave mentioned, would this get rid of the need of `cost_params.to_h.keys & required_params` ?
My main concern is code readability.  If your way is the best way, then I'm totally fine with it.

Reply:
The only thing my method actually has over params.require is that it allows me to form a response in my own format if the param is missing.